src/metabase/db/env.clj

```diff
@@ -43,101 +42,69 @@
 (def db-file
   "Path to our H2 DB file from env var or app config."
   ;; see https://h2database.com/html/features.html for explanation of options
-  (delay
-   (if (config/config-bool :mb-db-in-memory)
-     ;; In-memory (i.e. test) DB
-     ;; DB_CLOSE_DELAY=-1 = don't close the Database until the JVM shuts down
-     "mem:metabase;DB_CLOSE_DELAY=-1"
-     ;; File-based DB
-     (let [db-file-name (config/config-str :mb-db-file)]
-       (get-db-file db-file-name)))))
-
-(def ^:private jdbc-connection-regex
-  #"^(jdbc:)?([^:/@]+)://(?:([^:/@]+)(?::([^:@]+))?@)?([^:@]+)(?::(\d+))?/([^/?]+)(?:\?(.*))?$")
-
-(defn- parse-connection-string
-  "Parse a DB connection URI like
-  `postgres://cam@localhost.com:5432/cams_cool_db?ssl=true&sslfactory=org.postgresql.ssl.NonValidatingFactory` and
-  return a broken-out map."
-  [uri]
-  (when-let [[_ _ protocol user pass host port db query] (re-matches jdbc-connection-regex uri)]
-    (u/prog1 (merge {:type     (case (keyword protocol)
-                                 :postgres   :postgres
-                                 :postgresql :postgres
-                                 :mysql      :mysql
-                                 :h2         :h2)}
-
-                    (case (keyword protocol)
-                      :h2 {:db db}
-                      {:user     user
-                       :password pass
-                       :host     host
-                       :port     port
-                       :dbname   db})
-                    (some-> query
-                            codec/form-decode
-                            walk/keywordize-keys))
-      ;; If someone is using Postgres and specifies `ssl=true` they might need to specify `sslmode=require`. Let's let
-      ;; them know about that to make their lives a little easier. See https://github.com/metabase/metabase/issues/8908
-      ;; for more details.
-      (when (and (= (:type <>) :postgres)
-                 (= (:ssl <>) "true")
-                 (not (:sslmode <>)))
-        (log/warn (trs "Warning: Postgres connection string with `ssl=true` detected.")
-                  (trs "You may need to add `?sslmode=require` to your application DB connection string.")
-                  (trs "If Metabase fails to launch, please add it and try again.")
-                  (trs "See https://github.com/metabase/metabase/issues/8908 for more details."))))))
-
-(def ^:private connection-string-details
-  (delay (when-let [uri (config/config-str :mb-db-connection-uri)]
-           (parse-connection-string uri))))
+  (if (config/config-bool :mb-db-in-memory)
```

Comment:
IIRC the reason I put it there was so when you were launching the REPL it wouldn't crash if you had bad/malformed DB details. That was with our old Leiningen setup that basically initialized the entire system on launch. If your details were busted you'd have to try and fix them and then relaunch the whole thing which could take a minute or two each time.

Since we've switched to the Clojure CLI launching a REPL doesn't load any namespaces until you explicitly ask for them, so if this parses bad and causes loading to fail you can just fix it and reload the namespace